src/TaskLineRenderer.ts

```diff
@@ -0,0 +1,228 @@
+import { Component, MarkdownRenderer } from 'obsidian';
+import type { Moment } from 'moment';
+import type { Task } from './Task';
+import * as taskModule from './Task';
+import type { LayoutComponent, LayoutOptions } from './TaskLayout';
+import { TaskLayout } from './TaskLayout';
+import { replaceTaskWithTasks } from './File';
+import { getSettings } from './Config/Settings';
+
+export type TaskLineRenderDetails = {
+    parentUlElement: HTMLElement;
+    /** The nth item in this list (including non-tasks). */
+    listIndex: number;
+    layoutOptions?: LayoutOptions;
+    isFilenameUnique?: boolean;
+};
+
+/**
+ * The function used to render a Markdown task line into an existing HTML element.
+ */
+export type TextRenderer = (text: string, element: HTMLSpanElement, path: string) => Promise<void>;
+
+async function obsidianMarkdownRenderer(text: string, element: HTMLSpanElement, path: string) {
+    await MarkdownRenderer.renderMarkdown(text, element, path, null as unknown as Component);
+}
+
+/**
+ * Renders a given Task object into an HTML List Item (LI) element, using the given renderDetails
+ * configuration and a supplied TextRenderer (typically the Obsidian Markdown renderer, but for testing
+ * purposes it can be a simpler one).
+ * The element includes the task and its various components (description, priority, block link etc), the
+ * checkbox on the left with its event handling of completing the task, and the button for editing the task.
+ */
+export async function renderTaskLine(
+    task: Task,
+    renderDetails: TaskLineRenderDetails,
+    textRenderer: TextRenderer | null = null,
+): Promise<HTMLLIElement> {
+    if (!textRenderer) textRenderer = obsidianMarkdownRenderer;
+    const li: HTMLLIElement = document.createElement('li');
+    renderDetails.parentUlElement.appendChild(li);
+    li.classList.add('task-list-item', 'plugin-tasks-list-item');
+
+    const textSpan = document.createElement('span');
```

Comment:
In my initial notes I said:

> I don't know enough about rendering to know if this a change or not, but I am looking for things that might be changes in behaviour.
> I cannot find any reference to 'span' in the original code when I search for 6ad76001c2e0e9361219196c6809cf9094800081.
> Bit worried that this might break user CSS filters???
> Would anything need to be changed in https://github.com/obsidian-tasks-group/obsidian-tasks/blob/main/docs/advanced/styling.md ?

But when I ran the new version in Obsidian with the Developer Tools, I couldn't see any difference in what was generated...

So I think it's all OK...

Reply:
This is completely equivalent to the `const textSpan = li.createSpan();` line in `Task.toLi` on the main branch. The reason I changed this line to `document.createElement` is that `li.createSpan` is a shortcut in the Obsidian API, which the mock we use for running tests does not include, and I wanted the tests to correctly create the div and span structure. But it does exactly the same.